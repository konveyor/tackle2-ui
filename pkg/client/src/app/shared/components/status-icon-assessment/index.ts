export * from "./status-icon-assessment";
