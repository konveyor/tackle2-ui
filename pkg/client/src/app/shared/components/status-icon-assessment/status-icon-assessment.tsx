import React from "react";
import { useTranslation } from "react-i18next";

import { Flex, FlexItem, SpinnerProps } from "@patternfly/react-core";
import { CheckCircleIcon } from "@patternfly/react-icons/dist/esm/icons/check-circle-icon";
import { TimesCircleIcon } from "@patternfly/react-icons/dist/esm/icons/times-circle-icon";
import { InProgressIcon } from "@patternfly/react-icons/dist/esm/icons/in-progress-icon";
import { SVGIconProps } from "@patternfly/react-icons/dist/js/createIcon";
import {
  global_disabled_color_200 as disabledColor,
  global_success_color_100 as successColor,
  global_Color_dark_200 as unknownColor,
  global_info_color_200 as loadingColor,
} from "@patternfly/react-tokens";

export type StatusIconAssessmentType =
  | "NotStarted"
  | "InProgress"
  | "Completed";

type IconListType = {
  [key in StatusIconAssessmentType]: {
    Icon:
      | React.ComponentClass<SVGIconProps>
      | React.FunctionComponent<SpinnerProps>;
    color: { name: string; value: string; var: string };
  };
};
const iconList: IconListType = {
  NotStarted: {
    Icon: TimesCircleIcon,
    color: unknownColor,
  },
  InProgress: {
    Icon: InProgressIcon,
    color: loadingColor,
  },
  Completed: {
    Icon: CheckCircleIcon,
    color: successColor,
  },
};

export interface IStatusIconAssessmentProps {
  status: StatusIconAssessmentType;
  isDisabled?: boolean;
  className?: string;
}

export const StatusIconAssessment: React.FunctionComponent<
  IStatusIconAssessmentProps
> = ({
  status,
  isDisabled = false,
  className = "",
}: IStatusIconAssessmentProps) => {
  const { t } = useTranslation();

  const Icon = iconList[status].Icon;
  const icon = (
    <Icon
      color={isDisabled ? disabledColor.value : iconList[status].color.value}
      className={className}
    />
  );

  let label: string;
  switch (status) {
    case "NotStarted":
      label = t("terms.notStarted");
      break;
    case "InProgress":
      label = t("terms.inProgress");
      break;
    case "Completed":
      label = t("terms.completed");
      break;
    default:
      label = t("terms.unknown");
      break;
  }

  return (
    <Flex
      spaceItems={{ default: "spaceItemsSm" }}
      alignItems={{ default: "alignItemsCenter" }}
      flexWrap={{ default: "nowrap" }}
      style={{ whiteSpace: "nowrap" }}
      className={className}
    >
      <FlexItem>{icon}</FlexItem>
      <FlexItem>{label}</FlexItem>
    </Flex>
  );
};
