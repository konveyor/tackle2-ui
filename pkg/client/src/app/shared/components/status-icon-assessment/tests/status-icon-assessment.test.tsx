import React from "react";
import { shallow } from "enzyme";
import { StatusIconAssessment } from "../status-icon-assessment";

describe("StatusIconAssessment", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<StatusIconAssessment status="NotStarted" />);
    expect(wrapper).toMatchSnapshot();
  });
});
