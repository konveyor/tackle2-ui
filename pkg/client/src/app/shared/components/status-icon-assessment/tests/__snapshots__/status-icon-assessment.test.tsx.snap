// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIconAssessment Renders without crashing 1`] = `
<Flex
  alignItems={
    Object {
      "default": "alignItemsCenter",
    }
  }
  className=""
  flexWrap={
    Object {
      "default": "nowrap",
    }
  }
  spaceItems={
    Object {
      "default": "spaceItemsSm",
    }
  }
  style={
    Object {
      "whiteSpace": "nowrap",
    }
  }
>
  <FlexItem>
    <Component
      className=""
      color="#6a6e73"
    />
  </FlexItem>
  <FlexItem>
    terms.notStarted
  </FlexItem>
</Flex>
`;
